File: update_api_docs.php
<?php
ini_set('display_errors', 1);
error_reporting(E_ALL);

// Read the api.php docs file
$content = file_get_contents('docs/api.php');

if ($content === false) {
    die("Error reading file");
}

echo "Original file size: " . strlen($content) . " bytes\n";

// Regex pattern to match response blocks with 4 fields (without success/result/error)
// and replace with blocks that have all 6 fields
$pattern = '/("response_code": 200,\s*"message": "[^"]+",\s*"string": "[^"]*",\s*"language": "[^"]+",\s*"data": )(true|false)(\s*\})/m';

$replacement = '$1$2,
  "success": true,
  "result": $2,
  "error": null$3';

$newContent = preg_replace($pattern, $replacement, $content);

if ($newContent === null) {
    die("Regex error: " . preg_last_error());
}

$matches = preg_match_all($pattern, $content, $m);
echo "Found $matches response blocks to update\n";

// Specifically update the contains-all-logical-chars response string
$newContent = str_replace(
    '"message": "All logical characters check completed",
  "string": "hello",',
    '"message": "All logical characters check completed",
  "string": "అనన్య",',
    $newContent
);

echo "Updated file size: " . strlen($newContent) . " bytes\n";

if (file_put_contents('docs/api.php', $newContent) === false) {
    die("Error writing file");
}

echo "File updated successfully!\n";
echo "Total replacements made in response blocks\n";
?>

File: update_api_docs.py
#!/usr/bin/env python3
import re

# Read the file
with open('docs/api.php', 'r', encoding='utf-8') as f:
    content = f.read()

print(f"Original file size: {len(content)} bytes")

# Pattern to match response blocks that only have 4 fields
# and need success, result, and error added
pattern = r'("response_code": 200,\s*"message": "[^"]+",\s*"string": "[^"]*",\s*"language": "[^"]+",\s*"data": )(true|false)(\s*\})'

def add_response_fields(match):
    prefix = match.group(1)
    data_value = match.group(2)
    suffix = match.group(3)
    
    return f'{prefix}{data_value},\n  "success": true,\n  "result": {data_value},\n  "error": null{suffix}'

new_content = re.sub(pattern, add_response_fields, content, flags=re.MULTILINE)

print(f"Updated file size: {len(new_content)} bytes")

# Now fix specific string values for better examples
replacements = [
    ('All logical characters check completed",\n  "string": "hello",\n  "language": "telugu",',
     'All logical characters check completed",\n  "string": "అనన్య",\n  "language": "telugu",'),
]

for old, new in replacements:
    new_content = new_content.replace(old, new)
    if old in content:
        print(f"Updated: {old[:50]}...")

# Write the file back
with open('docs/api.php', 'w', encoding='utf-8') as f:
    f.write(new_content)

print("File updated successfully!")
